Yosemite/YosemiteTests/Stores/InboxNotesStoreTests.swift

```diff
@@ -0,0 +1,316 @@
+            self.store.onAction(action)
+        }
+
+
+        // Then the initial inbox note should have it's values updated
+        let updatedInboxNote = viewStorage.loadInboxNote(siteID: sampleSiteID, id: 296)
+        XCTAssertTrue(result.isSuccess)
+        XCTAssertNotEqual(initialInboxNote.name, updatedInboxNote?.name)
+        XCTAssertNotEqual(initialInboxNote.title, updatedInboxNote?.title)
+        XCTAssertNotEqual(initialInboxNote.content, updatedInboxNote?.content)
+    }
+
+    func test_loadAllInboxNotes_then_it_returns_error_upon_response_error() {
+        // Given a stubbed generic-error network response
+        network.simulateResponse(requestUrlSuffix: "admin/notes", filename: "generic_error")
+        XCTAssertEqual(storedInboxNotesCount, 0)
+
+        // When dispatching a `loadAllInboxNotes` action
+        let result: Result<[Networking.InboxNote], Error> = waitFor { [weak self] promise in
+            guard let self = self else {
+                return
+            }
+
+            let action = InboxNotesAction.loadAllInboxNotes(siteID: self.sampleSiteID) { result in
+                promise(result)
+            }
+            self.store.onAction(action)
+        }
+
+        // Then no inbox notes should be stored
+        XCTAssertEqual(storedInboxNotesCount, 0)
+        XCTAssertFalse(result.isSuccess)
+    }
+
+    func test_loadAllInboxNotes_then_it_returns_error_upon_empty_response() {
+        // Given an empty network response
+        XCTAssertEqual(storedInboxNotesCount, 0)
+
+        // When dispatching a `loadAllInboxNotes` action
+        let result: Result<[Networking.InboxNote], Error> = waitFor { [weak self] promise in
+            guard let self = self else {
+                return
+            }
+
+            let action = InboxNotesAction.loadAllInboxNotes(siteID: self.sampleSiteID) { result in
+                promise(result)
+            }
+            self.store.onAction(action)
+        }
+
+        // Then no inbox notes should be stored
+        XCTAssertEqual(storedInboxNotesCount, 0)
+        XCTAssertFalse(result.isSuccess)
+    }
+
+    func test_dismissInboxNote_then_it_update_inbox_note_upon_successful_response() {
+        // Given a stubbed inbox note network response
+        let sampleInboxNoteID: Int64 = 296
+        network.simulateResponse(requestUrlSuffix: "admin/notes/\(sampleInboxNoteID)", filename: "inbox-note")
+
+        // When dispatching a `dismissInboxNote` action
+        let result: Result<Networking.InboxNote, Error> = waitFor { [weak self] promise in
+            guard let self = self else {
+                return
+            }
+
+            let action = InboxNotesAction.dismissInboxNote(siteID: self.sampleSiteID, noteID: sampleInboxNoteID) { result in
+                promise(result)
+            }
+            self.store.onAction(action)
+        }
+
+        // Then a valid inbox note should be stored
+        XCTAssertEqual(storedInboxNotesCount, 1)
```

Comment:
nit: does any property change after dismissing a note? if so, maybe we can assert on that change as well?